actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -2,8 +2,14 @@ module ActionController
   module ImplicitRender
     def send_action(method, *args)
       ret = super
-      default_render unless performed?
-      ret
+      return ret if performed?
+
+      if template_exists?(method, _prefixes, variants: request.variant || [])
```

Comment:
`request.variant` always returns an `ArrayInquirer` now, and it will be empty if there are no variants. Don't think we need the fallback here.